lib/Optimizer/Transforms/ObserveAnsatz.cpp

```diff
@@ -90,6 +98,29 @@ struct AnsatzFunctionAnalysis {
         data.qubitValues.insert({op.getConstantIndex(), op.getResult()});
     });
 
+    // If mapping has moved qubits or introduced auxillary qubits, update the
+    // analysis accordingly.
+    if (auto mappingAttr =
+            dyn_cast_if_present<ArrayAttr>(funcOp->getAttr("mapping_v2p"))) {
+      // First populate mapping_v2p[].
+      SmallVector<int64_t> mapping_v2p(mappingAttr.size());
+      for (auto [origIx, mappedIx] : llvm::enumerate(mappingAttr)) {
+        if (auto val = dyn_cast<IntegerAttr>(mappedIx))
+          mapping_v2p[origIx] = val.getInt();
+        else
+          emitError(funcOp.getLoc(), "invalid mapping_v2p found in function");
+      }
+
+      // Next create newQubitValues[]
+      DenseMap<std::size_t, Value> newQubitValues;
```

Comment:
I'm a bit confused by the integer types that pop up. In this code there are both `int64_t` and `std::size_t` key values being used. That just feels a bit suspicious, I guess.